### PR TITLE
Add and implement sanity

### PR DESCRIPTION
**Pull request by PeterJFB**
Used for the content most prone to change:)

[Screencast from 06-16-2023 10:18:28 AM.webm](https://github.com/webkom/nyiabakus/assets/42469466/112c0009-84b8-4ffd-bffe-a8b62eec1298)

## Todo

- [x] Date-field from sanity is only comparing date, not month. Rewrite it (mayb reduce the spaghetti)

### CODE REVIEW

File: components/EventsListView/index.tsx [hunk 1]
```diff
@@ -22,28 +34,44 @@ const EventsListView: React.FC<EventsListViewProps> = ({ events }) => {
...
-            <EventItem key={event.id} event={event} />
-          ))}
-          {day.events.length === 0 && <p>Ingen arrangementer</p>}
+          {day.description ? (
+            <div className={styles.description}>
+              <CollapsibleItem title="" minHeight="50px">
+                <PortableText value={day.description.content} />
+              </CollapsibleItem>
+            </div>
+          ) : null}
```
Comment (norbye) [inline components/EventsListView/index.tsx:67]: `day.description && (...)`?

File: schemas/dayDescription.ts [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          (date ? ` ${date.split("-")[2]}.${date.split("-")[1]}` : ""),
+      };
+    },
+  },
+  fields: [
+    {
+      name: "dayOfWeek",
+      type: "string",
+      title: "Dag i uken",
+    },
```
Comment (norbye) [inline schemas/dayDescription.ts:25]: Can't this be extracted from the date as is done with the events?

File: pages/faq.tsx [hunk 3]
```diff
@@ -1,28 +1,8 @@
-import Head from "next/head";
 import { Inter } from "next/font/google";
-import styles from "@/styles/Home.module.css";
-import Header from "@/components/Header";
-import Navbar from "@/components/Navbar";
 import FaqContent from "@/components/FaqContent";
 
 const inter = Inter({ subsets: ["latin"] });
```
Comment (norbye) [inline pages/faq.tsx:4]: Delete this if unused? (As well as on the other pages)

File: pages/_app.tsx [hunk 4]
```diff
@@ -1,6 +1,22 @@
...
 export default function App({ Component, pageProps }: AppProps) {
-  return <Component {...pageProps} />;
+  return (
+    <>
+      <Head>
+        <title>Ny i Abakus</title>
+      </Head>
+      <main className={`${styles.main} ${inter.className}`}>
+        <Navbar />
+        <Component {...pageProps} />;
```
Comment (norbye) [inline pages/_app.tsx:18]: Yeet the semicolon

**Review by norbye — CHANGES_REQUESTED** [review]
Very nice:)) would be nice to get the semicolon out of the way before merging

File: package.json [hunk 5]
```diff
@@ -11,9 +11,18 @@
...
+    "@sanity/ui": "^1.6.0",
+    "@sanity/vision": "^3.12.1",
     "next": "13.3.1",
+    "next-sanity": "^4.3.3",
     "react": "18.2.0",
-    "react-dom": "18.2.0"
+    "react-dom": "18.2.0",
+    "react-portable-text": "^0.6.0",
+    "sanity": "^3.12.1",
+    "styled-components": "^5.3.9"
```
Comment (norbye) [inline package.json:25]: Is styled components used?

Comment (PeterJFB) [reply package.json:25]: yes by sanity studio :sweat_smile:

File: schemas/dayDescription.ts (hunk 2 above)
Comment (PeterJFB) [reply schemas/dayDescription.ts]: True, fixed:P

**Review by norbye — APPROVED** [review]

**Pull request merged**